sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/NonStreamingOrderByBadRequestException.java

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Can we reuse `BadRequestException`?

Reply:
We can, but using this exception would provide more context in the logs to folks who don't know about this new query pipeline.

Reply:
What about also using a unique sub status code? How did .Net solve this? If we use client-side substatus code it should be consistent across SDKs

Reply:
I am using this exception when we are checking for the top or limit + offset constraint. .Net doesn't has this constraint so they don't throw any exception in the logic. 

Python SDk is throwing a 'raise ValueError("")` which is equivalent to an IllegalArgumentException in java. 
I think we should stick with this one.